design/001-RPC-Interface-Specification.md

```diff
@@ -0,0 +1,413 @@
+
+1. Node APIs - For accessing node related functionality of the node. Following
+APIs should be provided:
+
+    1. Get config
+    2. Open Session
+    3. Help
+
+2. Session APIs - For session related functionality. Each request will include
+a Session ID. Following APIs should be provided:
+    1. Add contact
+    2. Get contacts
+    3. Open payment channel
+    4. Get payment channels
+    5. Subscribe to payment channel requests
+    6. Respond to payment channel request
+
+3. Channel APIS - For channel related functionality. Each request will include
+a Session ID and Channel ID.
+    1. Send payment
+    2. Request payment
+    3. Subscribe to payment channel updates
+    4. Respond to payment channel update
+    5. Get balance
+    6. Close channel
+
+### Data formats
+
+The following 4 data formats will be used in the API description.
+
+1. `Peer`
+
+    * `PeerAlias`: [String] Alias for the peer. It is a value set by the user to
+  reference the peer during API calls.
+    * `Perun ID`: [String] Permanent Identity of the peer in the off-chain
+  network.
+    * `Network Address`: [String] Network address for off-communications.
+    * `Protocol Versions`: [String] Perun protocol versions supported by the
+    peer.
+
+2. `Balance Info`
+
+    * `Currency`: [String] Currency used in the balance.
+    * List of `Alias`, `Balance` pairs, where
+      * `Alias`: [String] Alias of the peer represented in the channel. `Self` is
+        a special alias used for the user of the node.
+      * `Balance`: [String] Balance of the peer.
+
+3. `Currency String`: The balance of can be specified in currency string format
+  as show below. The letters used that can used in the currency string is
+determined by the value of currency field.
+
+    * 2e50c for currency "Euro" that represents 2 euros and 50 cents and,
+    * 2e50w for currency "Ethereum" that represents 2 ethers and 50 wei.
+
+4. `Channel`
+
+    * `Channel ID`: [String] Unique ID of the channel.
+    * `Channel Balance`: [Balance Info]
+    * `Version`: [Integer] Current Version of the channel.
+
+The following error messages will be used in the API description.
+
+1. `Unknown Session ID`: No session corresponding to the specified ID.
+2. `Unknown Channel ID`: No channel corresponding to the specified ID.
+3. `Unknown Subscription ID`: No subscription corresponding to the specified ID.
+4. `Peer Exists`: A contact entry exits for the given peer and cannot be
+5. `Peer Not Responding`: Peer did not respond within expected time period.
+
+### Node
+
+#### 1. Get Config
+
+Returns the configuration parameters of the node.
+
+*Parameters* none
+
+*Return*
+
+* `Node config`: (list of configurable parameters is yet to be defined)
+
+#### 2. Open Session
+
+Open a new session for the given user with the specified configuration file.
+
+*Parameters*
+
+* `Config File`: [String] Path to the config file for the session. This should
+   present on a file system path accesible by the node.
```

Comment:
Does it make sense also to pass the needed values instead of passing a file? As an alternative

Reply:
The following are the required parameters: on-chain credentials, off-chain credentials and database directory (for persisting data across sessions). Credentials include: address, path to keystore file and password.
and the following are optional parameters: chain URL, addresses of on-chain state channel contracts.

Since the password is included in the credentials (a required parameter), i think it is safe to specify it in a file and pass the path to it.

Maybe the addresses and optional parameters can be specified by the user over the API. However, since a file is required, I thought all the configuration can be provided via the file. 

Do you see benefits for passing part of the configuration via API ?

Reply:
You're right, I forgot the password. Then it will make only sense later if the password/signing is not handled by the node but an external secured service.

Reply:
Yes, When we add the support for making signatures via external service, then we could add an option to provide all values via API itself.